homeassistant/components/device_tracker/tile.py

```diff
@@ -34,89 +32,106 @@
+    """Define an object to retrieve Tile data."""
+
+    def __init__(self, client, hass, async_see, types, show_inactive):
         """Initialize."""
-        from pytile import Client
-
-        _LOGGER.debug('Received configuration data: %s', config)
-
-        # Load the client UUID (if it exists):
-        config_data = load_json(hass.config.path(CLIENT_UUID_CONFIG_FILE))
-        if config_data:
-            _LOGGER.debug('Using existing client UUID')
-            self._client = Client(
-                config[CONF_USERNAME],
-                config[CONF_PASSWORD],
-                config_data['client_uuid'])
-        else:
-            _LOGGER.debug('Generating new client UUID')
-            self._client = Client(
-                config[CONF_USERNAME],
-                config[CONF_PASSWORD])
-
-            if not save_json(
-                    hass.config.path(CLIENT_UUID_CONFIG_FILE),
-                    {'client_uuid': self._client.client_uuid}):
-                _LOGGER.error("Failed to save configuration file")
-
-        _LOGGER.debug('Client UUID: %s', self._client.client_uuid)
-        _LOGGER.debug('User UUID: %s', self._client.user_uuid)
-
-        self._show_inactive = config.get(CONF_SHOW_INACTIVE)
-        self._types = config.get(CONF_MONITORED_VARIABLES)
-
-        self.devices = {}
-        self.see = see
-
-        track_utc_time_change(
-            hass, self._update_info, second=range(0, 60, 30))
-
-        self._update_info()
-
-    def _update_info(self, now=None) -> None:
-        """Update the device info."""
-        self.devices = self._client.get_tiles(
-            type_whitelist=self._types, show_inactive=self._show_inactive)
-
-        if not self.devices:
-            _LOGGER.warning('No Tiles found')
-            return
-
-        for dev in self.devices:
-            dev_id = 'tile_{0}'.format(slugify(dev['name']))
-            lat = dev['tileState']['latitude']
-            lon = dev['tileState']['longitude']
-
-            attrs = {
-                ATTR_ALTITUDE: dev['tileState']['altitude'],
-                ATTR_CONNECTION_STATE: dev['tileState']['connection_state'],
-                ATTR_IS_DEAD: dev['is_dead'],
-                ATTR_IS_LOST: dev['tileState']['is_lost'],
-                ATTR_RING_STATE: dev['tileState']['ring_state'],
-                ATTR_VOIP_STATE: dev['tileState']['voip_state'],
-            }
-
-            self.see(
-                dev_id=dev_id,
-                gps=(lat, lon),
-                attributes=attrs,
-                icon=DEFAULT_ICON
-            )
+        self._async_see = async_see
+        self._client = client
+        self._hass = hass
+        self._show_inactive = show_inactive
+        self._types = types
+
+    async def async_init(self):
+        """Further initialize connection to the Tile servers."""
+        from pytile.errors import TileError
+
+        try:
+            await self._client.get_session()
+            await self._async_update()
```

Comment:
If there's a `TileError` inside `_async_update` it won't bubble up here since it's caught inside the method. Does that matter or are we only expecting `_client.get_session` to error?